connectors/src/api/webhooks/webhook_slack.ts

```diff
@@ -45,55 +46,110 @@ const _webhookSlackAPIHandler = async (
     return res.status(200).send({
       challenge: req.body.challenge,
     });
-  } else if (req.body.type === "event_callback") {
+  }
+
+  if (req.body.type === "event_callback") {
     if (!req.body.team_id) {
-      return res.status(400).send({
-        error: {
+      return apiError(req, res, {
+        api_error: {
+          type: "invalid_request_error",
           message: "Missing team_id in request body",
         },
+        status_code: 400,
       });
     }
-    const slackConfiguration = await SlackConfiguration.findOne({
+    const slackConfigurations = await SlackConfiguration.findAll({
       where: {
         slackTeamId: req.body.team_id,
       },
     });
-    if (!slackConfiguration) {
-      return res.status(404).send({
-        error: {
+    if (slackConfigurations.length === 0) {
+      return apiError(req, res, {
+        api_error: {
+          type: "connector_configuration_not_found",
           message: `Slack configuration not found for teamId ${req.body.team_id}`,
         },
+        status_code: 404,
       });
     }
 
-    if (req.body.event?.type === "message") {
-      if (req.body.team_id) {
-        let workflowRes = null;
-        if (req.body.event?.channel && req.body.event?.thread_ts) {
-          workflowRes = await launchSlackSyncOneThreadWorkflow(
-            slackConfiguration.connectorId.toString(),
-            req.body.event.channel,
-            req.body.event.thread_ts
+    switch (req.body.event?.type) {
+      /**
+       * `message` handler.
+       */
+      case "message": {
+        if (req.body.team_id) {
```

Comment:
Wrong test case, you should do `if (!req.body.team_id)`.
We are going to have to test this before pushing it to prod 💯

Reply:
Nice catch